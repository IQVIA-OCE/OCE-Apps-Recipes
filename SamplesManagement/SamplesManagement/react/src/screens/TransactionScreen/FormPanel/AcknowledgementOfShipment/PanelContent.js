import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select } from '@oce-apps/apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';
import { packageConditions } from '../../constants';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container} testID="AOSPanelContent">
      <View style={styles.col}>
        <DateField
          label="Received Date"
          value={values.fields.receivedDate}
          onChange={val => setFieldValue('fields.receivedDate', val)}
          style={styles.field}
          hasError={getFieldError('receivedDate', errors, touched)}
          helperText={getFieldHelperText('receivedDate', errors, touched)}
          required={readonly ? false : true}
          touched={touched}
          readonly={readonly}
        />
        {readonly ? <TextInput
          label="Condition of Package"
          value={values.fields.conditionOfPackage ? values.fields.conditionOfPackage.label : ''}
          fullWidth
          readonly={readonly}
        /> : <Select
          label="Condition of Package"
          placeholder={'-None-'}
          options={packageConditions}
          value={values.fields.conditionOfPackage}
          onChange={val => setFieldValue('fields.conditionOfPackage', val)}
          fullWidth
          style={{ width: '100%'}}
          error={getFieldError('conditionOfPackage', errors, touched)}
          helperText={getFieldHelperText('conditionOfPackage', errors, touched)}
          required
        />}

      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : null}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  required: {
    borderWidth: 0
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
