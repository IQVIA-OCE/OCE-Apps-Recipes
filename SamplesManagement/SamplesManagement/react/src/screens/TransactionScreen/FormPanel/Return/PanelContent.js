import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useFormikContext } from 'formik';
import { TextInput } from 'apollo-react-native';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container} testID="ReturnPanelContent">
      <View style={styles.col}>
        <DateField
          label="Shipment Date"
          value={values.fields.shipmentDate}
          onChange={val => setFieldValue('fields.shipmentDate', val)}
          style={styles.field}
          hasError={getFieldError('shipmentDate', errors, touched)}
          helperText={getFieldHelperText('shipmentDate', errors, touched)}
          touched={touched}
          readonly={readonly}
        />
        <TextInput
          label="Tracking Number"
          onChangeText={val => setFieldValue('fields.trackingNumber', val)}
          value={values.fields.trackingNumber}
          readonly={readonly}
          fullWidth
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Shipment Carrier"
          onChangeText={val => setFieldValue('fields.shipmentCarrier', val)}
          value={values.fields.shipmentCarrier}
          fullWidth
          style={styles.field}
          readonly={readonly}
        />
        <TextInput
          label="Comments"
          onChangeText={val => setFieldValue('fields.comments', val)}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
});

export default PanelContent;
