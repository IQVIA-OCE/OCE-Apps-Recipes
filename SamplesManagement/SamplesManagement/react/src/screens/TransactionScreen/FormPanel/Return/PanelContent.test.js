import React from 'react';
import PanelContent from './PanelContent';
import { useFormikContext } from 'formik';
import { TextInput } from '@oce-apps/apollo-react-native';
import DateField from '../DateField';
import { render, act, fireEvent } from '@testing-library/react-native';
import moment from 'moment';

jest.mock('formik');

const setFieldValue = jest.fn();
useFormikContext.mockReturnValue({
  values: {
    shipmentCarrier: '',
    comments: '',
    fields: {
      shipmentDate: '',
      trackingNumber: '',
    },
  },
  setFieldValue,
  errors: [],
  touched: false
});

describe('PanelContent', () => {
  beforeEach(() => {
    setFieldValue.mockClear();
  });

  it('should render properly', () => {
    const { getByText } = render(<PanelContent />);
    act(() => jest.runAllTimers());

    expect(getByText(/Tracking Number/)).toBeTruthy();
  });

  it('should call DateField onChange', () => {
    const date = moment().toISOString();
    const { UNSAFE_root } = render(<PanelContent />);
    act(() => jest.runAllTimers());

    fireEvent(UNSAFE_root.findByType(DateField), 'change', date);
    act(() => jest.runAllTimers());

    expect(setFieldValue).toHaveBeenCalledWith('fields.shipmentDate', date);
  });

  it('should call TextInputs onChange', () => {
    const { UNSAFE_root } = render(<PanelContent />);
    act(() => jest.runAllTimers());

    const [_, TrackingNumberInput, ShipmentCarrierInput, CommentInput] = UNSAFE_root.findAllByType(TextInput);
    fireEvent.changeText(TrackingNumberInput, 'test');
    expect(setFieldValue).toHaveBeenCalledWith('fields.trackingNumber', 'test');

    setFieldValue.mockClear();
    fireEvent.changeText(ShipmentCarrierInput, 'test');
    expect(setFieldValue).toHaveBeenCalledWith('fields.shipmentCarrier', 'test');

    setFieldValue.mockClear();
    fireEvent.changeText(CommentInput, 'test');
    expect(setFieldValue).toHaveBeenCalledWith('fields.comments', 'test');
  });
});
