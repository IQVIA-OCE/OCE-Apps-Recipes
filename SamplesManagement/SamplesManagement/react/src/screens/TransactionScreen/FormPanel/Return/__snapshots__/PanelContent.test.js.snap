// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContent should call DateField onChange 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "rgb(221, 219, 218)",
        "borderRightWidth": 1,
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "marginBottom": 15,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          Shipment Date
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onFocus={[Function]}
              paddingLeft={8}
              paddingRight={40}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 8,
                    "paddingRight": 40,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value="Invalid date"
            />
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 10,
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïƒ®
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Tracking Number
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Shipment Carrier
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 80,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PanelContent should call TextInputs onChange 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "rgb(221, 219, 218)",
        "borderRightWidth": 1,
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "marginBottom": 15,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          Shipment Date
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onFocus={[Function]}
              paddingLeft={8}
              paddingRight={40}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 8,
                    "paddingRight": 40,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 10,
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïƒ®
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Tracking Number
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="test"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Shipment Carrier
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="test"
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 80,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="test"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PanelContent should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "rgb(221, 219, 218)",
        "borderRightWidth": 1,
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "marginBottom": 15,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          Shipment Date
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onFocus={[Function]}
              paddingLeft={8}
              paddingRight={40}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 8,
                    "paddingRight": 40,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 10,
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïƒ®
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Tracking Number
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Shipment Carrier
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 40,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 80,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
